tests/python/contrib/test_arm_compute_lib/infrastructure.py

```diff
@@ -195,3 +202,24 @@ def verify_codegen(module, known_good_codegen, num_acl_modules,
             f"The JSON produced by codegen does not match the expected result. \n" \
             f"Actual={codegen_str} \n" \
             f"Expected={known_good_codegen_str}"
+
+
+def generate_trials(space, r_factor=3):
+    """Generate a list of trials given series of parameters."""
```

Comment:
This function still generates random workloads for testing, although the space is now fixed. I'd suggest two possible changes:

1. Change this function so that it will generate deterministic trials.

2. Improve the docstring of this function saying that it is non-deterministic and how it generates trials (i.e., algorithm). In addition, improve the `verify` to log complete information such as shapes and functions to make sure we can reproduce the failures in case.

Reply:
Thanks, these are good points. Since this change strictly shouldn't be part of this PR I think I'll revert this. The previous set of tests were based on a more deterministic approach - I was just concerned with the number of trials this could lead to. It's a good point about logging the info though, this was on my todo list but I can add that into this PR.

Reply:
Yes please add the logging part to this PR. I'm personally fine to leave `generate_trials` unchanged in this PR.